### DISCUSSION TITLE
Freyr - Stops and never resumes

### DISCUSSION THREAD

**Discussion by DcryptedCA** [RandomNinjaAtk/arr-scripts#96] [Q&A] [answered]
Hello,

In the middle of a download/processing Lidarr takes over and Freyr stops and never resumes:
 [•] Download Complete
 [•] Embedding Metadata...
  • [✓] 001 Name (Live West Hollywood, CA )
[Info] ImportDecisionMaker: Reading file 1/10 
[Info] ImportDecisionMaker: Reading file 2/10 
[Info] ImportDecisionMaker: Reading file 3/10 
[Info] ImportDecisionMaker: Reading file 4/10 
[Info] ImportDecisionMaker: Reading file 5/10 ..

The log is always busy before Lidarr steps in and only way to resume Freyr is restarting the docker. 
What's the solution to this?

Thanks,

--Phil

**Comment by RandomNinjaAtk** [comment]
You should look at the script log to see what is going on.... You would also need to increase the lidarr logging to see what Lidarr is doing...

As for client issues, can't really help with that...

**Reply by DcryptedCA** [reply]
Which log shows Freyr history? 
To be clear Freyr should stop when Lidarr is performing an action? Should it resume (if it should stop) after that action?

Thanks,

--Phil

**Reply by DcryptedCA** [reply]
I ran the Audio script on it's own and it does stop and resume properly - however it did get stuck on the first album when collecting metadata so I ran the same command on a Dockerized Freyr and it worked fine. Once I bypassed the first issue album the script (Audio) runs fine on it's own, I just need to see again when starting the Lidarr+extended docker if it acts the same.

Thanks,

--Phil

**Comment by DcryptedCA** [comment]
Ok, there is definitely something wrong with the Freyr inside the extended script. With two different artists it freezes when Embedding Metadata, however if I use my Freyr only docker it does not have this issue with the exact same album when copied (which implies it's not a Freyr specific problem).

**Reply by RandomNinjaAtk** [reply]
You can see below that the script just hands off the request to the client, so it's an issue with the client and maybe whatever particular album it's trying to pull...

https://github.com/RandomNinjaAtk/arr-scripts/blob/78240a015f110a5419a27e68cba3f5035e09297a/lidarr/Audio.service.bash#L154

Additionally you can look at the running process (`ps aux`) and  you should be able to locate and kill it if needed...

**Comment by DcryptedCA** [comment]
Got this error now:
```
:: Search Complete...
jq: error (at <stdin>:1): Cannot iterate over null (null)
jq: error (at <stdin>:1): Cannot iterate over null (null)
jq: error (at <stdin>:4): Cannot iterate over null (null)
jq: error (at <stdin>:4): Cannot iterate over null (null)
jq: error (at <stdin>:4): Cannot iterate over null (null)
Audio: line 1156: 20231023 -  : syntax error: operand expected (error token is "-  ")
root@e777fbf8bea7:/config/custom-services.d#  
```

Restarted it in console and then it froze in another metadata  search 
"deezer:album:304262457"

Checked PS AUX (kill -9 killed the 3 freyr, then it restarted in the original instance):

```
root         1  0.0  0.0    208    24 ?        Ss   Oct22   0:00 /package/admin/s6/command/s6-svscan -d4 -- /run/service
root        15  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise s6-linux-init-shutdownd
root        17  0.0  0.0    200     4 ?        Ss   Oct22   0:00 /package/admin/s6-linux-init/command/s6-linux-init-shutdownd -c /run/s6/basedir -g 3000 -C -B
root        90  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise s6rc-oneshot-runner
root        91  0.0  0.0    212    16 ?        S    Oct22   0:00 s6-supervise s6rc-fdholder
root        92  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise custom-svc-Audio
root        93  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise custom-svc-AutoArtistAdder
root        94  0.0  0.0    212    24 ?        S    Oct22   0:00 s6-supervise custom-svc-AutoConfig
root        95  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise custom-svc-QueueCleaner
root        96  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise custom-svc-TidalVideoDownloader
root        97  0.0  0.0    212    24 ?        S    Oct22   0:00 s6-supervise custom-svc-UnmappedFilesCleaner
root        98  0.0  0.0    212    20 ?        S    Oct22   0:00 s6-supervise custom-svc-Video
root        99  0.0  0.0    212    24 ?        S    Oct22   0:00 s6-supervise svc-lidarr
root       107  0.0  0.0    188     4 ?        Ss   Oct22   0:00 /package/admin/s6/command/s6-ipcserverd -1 -- /package/admin/s6/command/s6-ipcserver-access -v0 -E -l0 -i data/rules -- /package/admin/s6/command
root       400  0.0  0.0   2416   524 ?        Ss   Oct22   0:00 bash ./run custom-svc-AutoConfig
root       402  0.0  0.0   2884   972 ?        Ss   Oct22   0:00 bash ./run custom-svc-Video
root       403  0.0  0.0   2808   912 ?        Ss   Oct22   0:00 bash ./run custom-svc-TidalVideoDownloader
root       405  0.0  0.0   2396   516 ?        Ss   Oct22   0:00 bash ./run custom-svc-UnmappedFilesCleaner
root       406  0.0  0.0  10072  8208 ?        Ss   Oct22   0:52 bash ./run custom-svc-Audio
abc        407  7.1  0.7 2912920 241548 ?      Ssl  Oct22 133:40 /app/lidarr/bin/Lidarr -nobrowser -data=/config
root       408  0.0  0.0   2608  1296 ?        Ss   Oct22   0:01 bash ./run custom-svc-QueueCleaner
root       409  0.0  0.0   2580   668 ?        Ss   Oct22   0:00 bash ./run custom-svc-AutoArtistAdder
root       521  0.0  0.0   2220    56 ?        S    Oct22   0:00 sleep infinity
root       531  0.0  0.0   2220    52 ?        S    Oct22   0:00 sleep infinity
root       536  0.0  0.0   2220    56 ?        S    Oct22   0:00 sleep infinity
root       547  0.0  0.0   2220    56 ?        S    Oct22   0:00 sleep infinity
root       816  0.0  0.0   2220    56 ?        S    Oct22   0:00 sleep infinity
root      1243  0.2  0.0  10168  9224 pts/1    S+   11:05   0:03 bash Audio
root     14662  0.0  0.0   9740  7928 pts/0    S+   Oct22   0:00 bash Audio
root     15495  0.2  2.1 82556252 700788 pts/0 Sl+  Oct22   3:35 node /usr/local/bin/freyr --no-bar -d /config/extended/downloads/audio/incomplete deezer:album:76834902
root     15496  0.0  0.0   2224    52 pts/0    S+   Oct22   0:00 tee -a /config/logs/Audio-2023_10_22_11_58_AM.txt
root     15554  0.0  0.0   4580   536 pts/0    S+   Oct22   0:00 AtomicParsley /config/extended/downloads/audio/incomplete/Name/Name/Name.
root     15729  0.0  0.0   2600   988 pts/1    Ss   Oct22   0:00 bash
root     24545  7.0  0.5 22316100 186944 pts/1 Sl+  11:27   0:27 node /usr/local/bin/freyr --no-bar -d /config/extended/downloads/audio/incomplete deezer:album:304262457
root     24546  0.0  0.0   2224   584 pts/1    S+   11:27   0:00 tee -a /config/logs/Audio-2023_10_23_11_05_AM.txt
root     24560  0.0  0.0   4536  2068 pts/1    S+   11:27   0:00 AtomicParsley /config/extended/downloads/audio/incomplete/Name/Name/Name.m4a 
root     24616  0.0  0.0   2220   652 ?        S    11:30   0:00 sleep 15m
root     24622  0.0  0.0   2600  2300 pts/2    Ss   11:31   0:00 bash
root     24663  0.0  0.0   2536  1712 pts/2    R+   11:33   0:00 ps aux
root     26436  0.1  1.5 79751688 510264 ?     Sl   Oct22   2:17 node /usr/local/bin/freyr --no-bar -d /config/extended/downloads/audio/incomplete deezer:album:212694892
root     26437  0.0  0.0   2224    56 ?        S    Oct22   0:00 tee -a /config/logs/Audio-2023_10_22_04_29_AM.txt
root     26476  0.0  0.0   4588   544 ?        S    Oct22   0:00 AtomicParsley /config/extended/downloads/audio/incomplete/Name/Name/Name.m4a --overWrite  --title Name
root     27122  0.0  0.0   2600   792 pts/0    Ss   Oct22   0:00 bash
```

I'll keep an eye on it to see what it is hanging on, not seeing any log reports that show any reason for the stoppage, but almost like it cannot download the metadata and then cannot continue but why it does when run in the Freyr docker, maybe it's a different revision?

**Reply by RandomNinjaAtk** [reply]
Since the file name/path is `/Name/Name/Name.m4a`, maybe the client is having difficulty pulling the data from online... Because that seems to indicate it doesn't have the info needed because its just generic `Name`....

I am however, just speculating as I don't really know and it's just what I suspect.... Again though, not sure what can be done to resolve it...

When you are testing, are you also performing your tests within the container?

It's also possible that maybe your hitting some sort of throttling issue that is preventing it from being able to communicate....

These are just thoughts....

**Comment by DcryptedCA** [comment]
The naming was just me changing from the log (didn't want to copy in artist names lol).
I am testing within the container (lidarr + extended). 
Just now I confirmed that when the Freyr script doesn't resume, if I kill it it resumes right away. Still not sure why it freezes on Metadata and doesn't resume automatically after Lidarr tasks are complete.

**Reply by DcryptedCA** [reply, marked as answer]
I run this script every 5 minutes now (In user scripts / Unraid), it kills Freyr inside the Lidarr docker and continues it where it left off if stuck or if it never resumed. It is constant now!
```
#!/bin/bash

export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

# Find the container ID based on the container name 
container_name="lidarr"
container_id=$(docker ps -qf "name=$container_name")

# Check if container ID was found
if [ -z "$container_id" ]; then
    echo "Container with name '$container_name' not found."
    exit 1
fi

process_command="node /usr/local/bin/freyr"

# Find the PID of the process based on the command within the found container
pid=$(docker exec "$container_id" pgrep -o -f "$process_command")

# Check if pgrep command encountered an error
if [ $? -ne 0 ]; then
    echo "Error running pgrep. Check if Docker and the container are accessible."
    exit 1
fi

# Check if a PID was found
if [ -n "$pid" ]; then
    echo "Found PID: $pid"
    # Kill the process with a -9 signal
    docker exec "$container_id" kill -9 "$pid"

    # Check if the kill command encountered an error
    if [ $? -eq 0 ]; then
        echo "Process with PID $pid has been killed."
    else
        echo "Error killing the process with PID $pid."
        exit 1
    fi
else
    echo "Process with command '$process_command' not found in the container."
fi

exit 0

```
